packages/Chem/src/utils/locked-entitie.ts

```diff
@@ -0,0 +1,38 @@
+export class LockedEntity<T extends any> {
```

Comment:
Did we implement a semaphore?

https://gist.github.com/cahilfoley/4b1b2f3fa9e2f9652ee1d8501443b5ca

Reply:
not really, that one I think is for limiting number of operations to the `max `number at one time. ours makes sure that not more than one 'thread' accesses or increments the index pointer at one time